File: CS/ColumnBase.Binding/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace ColumnBase.Binding.ViewModels {
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<IBaseItem> Items { get => GetProperty(() => Items); set => SetProperty(() => Items, value); }

        public MainViewModel() {
            Items = new ObservableCollection<IBaseItem>(Enumerable.Range(0, 10).Select(i =>
                new Item { Id = i, Name = $"Item {i}", CreatedAt = DateTime.Today.AddDays(i) }));
        }
    }

    public interface IBaseItem {
        DateTime CreatedAt { get; set; }
    }

    public interface IItem : IBaseItem {
        int Id { get; set; }
        string Name { get; set; }
    }

    public class Item : BindableBase, IItem {
        public DateTime CreatedAt { get => GetProperty(() => CreatedAt); set => SetProperty(() => CreatedAt, value); }

        public int Id { get => GetProperty(() => Id); set => SetProperty(() => Id, value); }

        public string Name { get => GetProperty(() => Name); set => SetProperty(() => Name, value); }
    }
}

File: VB/ColumnBase.Binding/ViewModels/MainViewModel.vb
Imports System.Collections.ObjectModel
Imports System.Linq
Imports DevExpress.Mvvm.CodeGenerators

Namespace ColumnBase.Binding.ViewModels

    <GenerateViewModel>
    Public Partial Class MainViewModel

        <GenerateProperty>
        Private _items As ObservableCollection(Of IBaseItem)

        Public Sub New()
            _items = New ObservableCollection(Of IBaseItem)(Enumerable.Range(0, 10).[Select](Function(i) New Item With {.Id = i, .Name = $"Item {i}", .CreatedAt = Date.Today.AddDays(i)}))
        End Sub
    End Class

    Public Interface IBaseItem

        Property CreatedAt As Date

    End Interface

    Public Interface IItem
        Inherits IBaseItem

        Property Id As Integer

        Property Name As String

    End Interface

    <GenerateViewModel>
    Public Partial Class Item
        Implements IItem

        <GenerateProperty>
        Private _createdAt As Date

        <GenerateProperty>
        Private _id As Integer

        <GenerateProperty>
        Private _name As String
    End Class
End Namespace
